x/tss/abci.go

```diff
@@ -1,293 +1,46 @@
-					oldKeyIDs, err := keeper.GetOldActiveKeyIDs(ctx, chain, role)
-					if err != nil {
-						keeper.Logger(ctx).Error(fmt.Sprintf("unable to retrieve old keys for chain %s with role %s: %s",
-							chain.Name, role.SimpleString(), err))
-						continue
-					}
-
-					for _, keyID := range oldKeyIDs {
-						keyInfos = append(keyInfos, types.KeyInfo{KeyID: keyID, KeyType: keyType})
-					}
-				}
-			}
-		}
-
-		bz := types.ModuleCdc.LegacyAmino.MustMarshalJSON(keyInfos)
-		ctx.EventManager().EmitEvent(sdk.NewEvent(types.EventTypeHeartBeat,
-			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
-			sdk.NewAttribute(sdk.AttributeKeyAction, types.AttributeValueSend),
-			sdk.NewAttribute(types.AttributeKeyKeyInfos, string(bz)),
-		))
+		emitHeartbeatEvent(ctx, m, n)
 	}
 
+	return nil
 }
 
-// sequentialSign limits tss sign within max signing shares
-func sequentialSign(ctx sdk.Context, signQueue utils.SequenceKVQueue, k types.TSSKeeper, s types.Snapshotter, voter types.Voter) {
-	i := uint64(0)
-	signShares := int64(0)
-	var signInfo exported.SignInfo
-
-	defer func() {
-		k.Logger(ctx).Debug(fmt.Sprintf("%d active sign shares, %d signatures in queue", signShares, signQueue.Size()))
-	}()
-
-	maxSignShares := k.GetMaxSimultaneousSignShares(ctx)
-	for signShares < maxSignShares && signQueue.Peek(i, &signInfo) {
-		_, sigStatus := k.GetSig(ctx, signInfo.SigID)
-		// no need to check if snapshot exists again, sanity check for that passed at this point
-		snap, _ := s.GetSnapshot(ctx, signInfo.SnapshotCounter)
+func emitHeartbeatEvent(ctx sdk.Context, m types.MultiSigKeeper, n types.Nexus) {
+	var keyInfos []types.KeyInfo
 
-		switch sigStatus {
-		case exported.SigStatus_Queued:
-			signShares += snap.CorruptionThreshold + 1
-			if signShares > maxSignShares {
-				return
-			}
-			emitSignStartEvent(ctx, k, voter, signInfo, snap)
-			k.SetInfoForSig(ctx, signInfo.SigID, signInfo)
-			k.SetSigStatus(ctx, signInfo.SigID, exported.SigStatus_Signing)
-			k.Logger(ctx).Debug(fmt.Sprintf("starting sign %s", signInfo.SigID))
-			i++
-		case exported.SigStatus_Signing:
-			signShares += snap.CorruptionThreshold + 1
-			k.Logger(ctx).Debug(fmt.Sprintf("signing %s in progress", signInfo.SigID))
-			i++
-		case exported.SigStatus_Signed, exported.SigStatus_Aborted, exported.SigStatus_Invalid:
-			signQueue.Dequeue(i, &signInfo)
-			k.Logger(ctx).Debug(fmt.Sprintf("dequeque %s, sign status %s", signInfo.SigID, sigStatus))
-		default:
-			panic("invalid sig status type")
-		}
-	}
-}
-
-// request proxies to initiate a tss signing protocol using the specified signing metadata
-func emitSignStartEvent(ctx sdk.Context, k types.TSSKeeper, voter types.InitPoller, info exported.SignInfo, snap snapshot.Snapshot) {
-	var nonParticipantShareCounts []int64
-	var nonParticipants []string
+	for _, chain := range n.GetChains(ctx) {
+		keyIDs := m.GetActiveKeyIDs(ctx, chain.Name)
```

Comment:
For cosmos chains, there are no active key ids. This should return an empty list, but I'd be happier if there's a test for that scenario for `GetActiveKeyIDs`

Reply:
Added one